### DISCUSSION TITLE
One rate per Customer, variable internal rates per user (in Kimai v1)

### DISCUSSION THREAD

**Discussion by hyssop** [kimai/kimai#4012] [Questions & Answers] [answered]
Note: this is in Kimai v1

My customers have one rate per customer, but my employees have different internal rates. I want to control each rate in ONE location, but I can't get that to work. Instead I have to set a user rate for EACH employee in EVERY customer. Then when I give an employee a raise, I have to re-do their internal rate for EACH customer. This is not efficient.

What want to do:
1- set a rate for each customer (but no internal rate, currently not possible)
2- set an internal rate for each user preference (leaving their rate $0, possible)
3- the desired result being that the customer rate will override the $0 user rate (this works), AND the user's internal rate will override the customer's internal rate (this doesn't work because leaving the customer internal rate empty causes it to default to the rate and the user internal rate is ignored)

Is this configuration possible and am I just missing something?

The current logic is that a user rate is used if there is no applicable customer/project/activity rate, and I assume the same for internal rates. But there is no way to set a customer rate without setting an internal rate also (internal rate defaults to rate if no internal rate is specified), thus the scenario I want to set up results in the internal rate being the customer rate, unless I add each user's rates to each customer. 

Could there be an option added to not default the customer/project/activity internal rate to the rate, if internal rate is left blank? If I understand the logic correctly that would seem to solve the issue. Either that, or an option to have higher user rates override lower customer/project/activity rate (n/a to fixed rates).

Thoughts, anyone?

-Pete

**Comment by kevinpapst** [comment, marked as answer]
What you describe has been implemented in 2.0.
But no feature will be backported to 1.x, so you have to upgrade. 
Probably you want to verify in the demo, that it does what you want.

**Reply by hyssop** [reply]
> What you describe has been implemented in 2.0. 
> But no feature will be backported to 1.x, so you have to upgrade. 
> Probably you want to verify in the demo, that it does what you want.

I read that v2 is basically ready for production now, but just waiting on plugins to be ready.
I currently use no plugins, so is there any reason not to update now?

**Reply by kevinpapst** [reply]
No, and the plugins were released this week. It is just me, who needs to write some news about it.

**Reply by BeckeBauer** [reply]
> No, and the plugins were released this week. It is just me, who needs to write some news about it.

Yes, please do 🙏 (can't wait to update)
